.github/workflows/ci.yml

```diff
@@ -2,157 +2,82 @@ name: PIC CI
-        run: |
-          rm -rf build
-          mkdir build && cd build
-          cmake .. -DBUILD_TEST=TRUE -DALIGNED_MEMORY_MODEL=TRUE -DFIXUP_ANNEX_B_TRAILING_NALU_ZERO=TRUE
-          make
-      - name: Run tests
-        run: |
-          cd build
-          ./tst/kvspic_test --gtest_filter="-TimerQueueFunctionalityTest.*:HeapPerfTest.*"
-  mac-os-m1-build-clang:
-    runs-on: macos-13-xlarge
-    env:
-      AWS_KVS_LOG_LEVEL: 2
-    steps:
-      - name: Clone repository
-        uses: actions/checkout@v2
-      - name: Build repository
-        run: |
-          brew unlink openssl
-          mkdir build && cd build
-          cmake .. -DBUILD_TEST=TRUE -DCMAKE_C_COMPILER=$(brew --prefix llvm@15)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm@15)/bin/clang++
-          make
-      - name: Run tests
-        run: |
-          cd build
-          ./tst/kvspic_test --gtest_filter="-TimerQueueFunctionalityTest.*:HeapPerfTest.*"
-      - name: Build With ALIGNED_MEMORY_MODEL and FIXUP_ANNEX_B_TRAILING_NALU_ZERO definitions TRUE
-        run: |
-          rm -rf build
-          mkdir build && cd build
-          cmake .. -DBUILD_TEST=TRUE -DALIGNED_MEMORY_MODEL=TRUE -DFIXUP_ANNEX_B_TRAILING_NALU_ZERO=TRUE -DCMAKE_C_COMPILER=$(brew --prefix llvm@15)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm@15)/bin/clang++
-          make
-      - name: Run tests
-        run: |
-          cd build
-          ./tst/kvspic_test --gtest_filter="-TimerQueueFunctionalityTest.*:HeapPerfTest.*"
-  mac-os-m1-build-gcc:
-    runs-on: macos-13-xlarge
-    env:
-      CC: /opt/homebrew/bin/gcc-13
-      CXX: /opt/homebrew/bin/g++-13
-      AWS_KVS_LOG_LEVEL: 2
+  mac-tests:
+    strategy:
+      matrix:
+        os: [ macos-13, macos-15 ]
+        config:
+          - name: Default Build
+            cmake_flags: "-DBUILD_TEST=TRUE"
+          - name: Aligned Memory Model Build
+            cmake_flags: "-DBUILD_TEST=TRUE -DALIGNED_MEMORY_MODEL=TRUE -DFIXUP_ANNEX_B_TRAILING_NALU_ZERO=TRUE"
+        compiler: [ gcc, clang ]
+      fail-fast: false
+
+    runs-on: ${{ matrix.os }}
+
     steps:
-      - name: Setup XCode version 15.2
-        uses: maxim-lobanov/setup-xcode@v1
-        with:
-          xcode-version: '15.2'
-      - name: Clone repository
-        uses: actions/checkout@v2
-      - name: Build repository
-        run: |
-          brew unlink openssl
-          mkdir build && cd build
-          cmake .. -DBUILD_TEST=TRUE
-          make
-      - name: Run tests
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up compiler
         run: |
+          if [[ "${{ matrix.compiler }}" == "gcc" ]]; then
+            brew install gcc
+          else
+            brew install llvm
+          fi
+
+      - name: Configure and build ${{ matrix.config.name }}
+        run: |
+          if [[ "${{ matrix.compiler }}" == "gcc" ]]; then
+            export CC="gcc"
+            export CXX="g++"
```

Comment:
For this to actually use the GNU compiler, we need to tag on the version, any of these will work:
```
gcc-12
gcc-13
gcc-14
```

Reply:
Adjusted to use 14 (latest) as does the clang.